api/internalutils/stream/stream.go

```diff
@@ -186,6 +186,46 @@ func MapWhile[A, B any](stream Stream[A], fn func(A) (B, bool)) Stream[B] {
 	}
 }
 
+// chain is a stream that performs a Chain operation.
+type chain[T any] struct {
+	streams []Stream[T]
+	err     error
+}
+
+func (stream *chain[T]) Next() bool {
+	for len(stream.streams) != 0 && stream.err == nil {
+		if stream.streams[0].Next() {
+			return true
+		}
+
+		stream.err = stream.streams[0].Done()
+		stream.streams[0] = nil
+		stream.streams = stream.streams[1:]
+	}
+
+	return false
+}
+
+func (stream *chain[T]) Item() T {
+	return stream.streams[0].Item()
```

Comment:
Yes, but `Next()` would have returned false in that case. All of the stream combinators that are slice-based (e.g. `PageFunc`, `Slice`) panic if `Item()` is called after `Next()` already returned `false` (or before the first call to `Next()`).  And the `Empty()` combinator _always_ panics if `Item()` is called.  It's a well documented part of the `Stream` interface that `Item()` is allowed to panic if the return value of `Next()` is not respected, since `Next()` is in part intended to be a bounds-check.